spec/controllers/todos_controller_spec.rb

```diff
@@ -45,5 +61,34 @@
 		  post :create, todo: {description: "I like cooking!"}
 		  Todo.count.should == 0
 		end
+		it "does not create tags without inline locations" do
+			post :create, todo: {name: "cook"}
+		  Tag.count.should == 0
+		end
+		it "does not create tags with at in a word without inline locations" do
+			post :create, todo: {name: "eat an apple"}
+		  Tag.count.should == 0
+		end
+		it "create a tag with upcase AT" do
+			post :create, todo: {name: "shop AT the Apple Store"}
+		  Tag.all.map(&:name).should == ["location:the Apple Store"]
+		end
+		context "with inline locations" do
+			it "creates a tag with one location" do
+				post :create, todo: {name: "cook AT home"}
+				Tag.all.map(&:name).should ==['location:home']
+			end
+			it "creates two tags with two locations" do
+				post :create, todo: {name: "cook AT home and work"}
+				Tag.all.map(&:name).should ==['location:home', 'location:work']
+			end
+			it "creates mutiple tags with mutiple locations" do
+				post :create, todo: {name: "cook AT home, work, school and library"}
+				Tag.all.map(&:name).should ==['location:home', 'location:work', 'location:school', 'location:library']
+			end
+			it_behaves_like "require_sign_in" do
+				let(:action) {post :create, todo: {name: "something"}}
+			end
+		end
```

Comment:
The indentation needs to be fixed in this whole file. You have to care about how your code looks on GitHub for review.